From: Heiko Carstens <heiko.carstens@de.ibm.com>
Subject: net: initcall order
References: 154669, IBM Codestream linux-2.6.16 october2005, patch 02-17

If the network code is compiled into the kernel the initcalls of all network
drivers are done before inet_init(). This is the wrong order since the drivers
rely on data structures that get initialized from inet_init().
Raise priority of inet_init() initcall from device_initcall() to fs_initcall().

Acked-by: Jan Blunck <jblunck@suse.de>
---
 net/ipv4/af_inet.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.16/net/ipv4/af_inet.c
===================================================================
--- linux-2.6.16.orig/net/ipv4/af_inet.c
+++ linux-2.6.16/net/ipv4/af_inet.c
@@ -1245,7 +1245,7 @@ out_unregister_udp_proto:
 	goto out;
 }
 
-module_init(inet_init);
+fs_initcall(inet_init);
 
 /* ------------------------------------------------------------------------ */
 
